plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1103,16 +1103,20 @@ public void testArithmeticPredicatePushdown()
                 .isFullyPushedDown()
                 .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
 
-        assertThatThrownBy(() -> query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2"))
-                .hasMessageContaining("by zero");
+        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2", errorMessageForDivisionByZeroPushdown());
 
         // Expression that evaluates to 0 for some rows on RHS of modulus
-        assertThatThrownBy(() -> query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % (regionkey - 1) = 2"))
-                .hasMessageContaining("by zero");
+        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % (regionkey - 1) = 2", errorMessageForDivisionByZeroPushdown());
 
         // TODO add coverage for other arithmetic pushdowns https://github.com/trinodb/trino/issues/14808
     }
 
+    @Language("RegExp")
+    protected String errorMessageForDivisionByZeroPushdown()
```

Comment:
We usually let the subclass inspect the exception object

https://github.com/trinodb/trino/blob/a480bd7ad57c35509fdb3d8838d681663795c2c3/testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java#L2004-L2006

Reply:
```java
assertThatThrownBy(() -> query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2"))
                .satisfies(this::verifyDivisionByZeroFailure);
```

and then
```java
    protected void verifyDivisionByZeroFailure(Throwable e)
    {
        throw new UnsupportedOperationException("This method should be overridden");
    }
```